b2b_ecommerce/api.py

```diff
@@ -0,0 +1,92 @@
+"""Functions for b2b_ecommerce"""
+from decimal import Decimal
+import uuid
+
+from django.conf import settings
+from django.db import transaction
+
+from ecommerce.api import create_coupons, ISO_8601_FORMAT, sign_cybersource_payload
+from ecommerce.models import CouponPaymentVersion
+from mitxpro.utils import now_in_utc
+
+
+def complete_b2b_order(order):
+    """
+    Create the enrollment codes which were paid for and link them to the purchaser.
+
+    Args:
+        order (B2BOrder): A fulfilled order for enrollment codes
+    """
+    with transaction.atomic():
+        product_id = order.product_version.product.id
+        payment_version = create_coupons(
+            name=f"CouponPayment for order #{order.id}",
+            product_ids=[product_id],
+            amount=Decimal("1"),
+            num_coupon_codes=order.num_seats,
+            coupon_type=CouponPaymentVersion.SINGLE_USE,
+        )
+        order.coupon_payment_version = payment_version
+        order.save()
```

Comment:
`save_and_log`?

Reply:
I wanted to hold off on `save_and_log` until the order fulfillment happens in the view functions so that there is only one record of change instead of multiple records in the audit table. But I do need to move `save_and_log` in `B2BCheckoutView` to be after this function so that it captures the output correctly.